overlord/devicestate/firstboot.go

```diff
@@ -92,15 +91,13 @@ func populateStateFromSeedImpl(st *state.State) ([]*state.TaskSet, error) {
 
 	// ack all initial assertions
 	model, err := importAssertionsFromSeed(st)
-	if err == errNothingToDo {
-		return trivialSeeding(st, markSeeded), nil
-	}
 	if err != nil {
 		return nil, err
 	}
 
+	assertSeedDir := filepath.Join(dirs.SnapSeedDir, "assertions")
 	seedYamlFile := filepath.Join(dirs.SnapSeedDir, "seed.yaml")
-	if release.OnClassic && !osutil.FileExists(seedYamlFile) {
+	if release.OnClassic && (!osutil.FileExists(seedYamlFile) || !osutil.FileExists(assertSeedDir)) {
```

Comment:
having seed.yaml without an explicit model file is intentionally not supported

Reply:
mmh, this code preserve that but I find the new order of checks not easy to parse